bottom/protocol.py

```diff
@@ -1,29 +1,38 @@
 import asyncio
 import logging
 from bottom.unpack import unpack_command
+from typing import Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from bottom.client import Client  # noqa
+
 # Always write the full \r\n per spec, but accept \n when reading
 DELIM = b"\r\n"
 DELIM_COMPAT = b"\n"
 log = logging.getLogger('bottom')
 
 
 class Protocol(asyncio.Protocol):
-    def __init__(self, client=None):
-        self.client = client
-        self.closed = False
-        self.transport = None
-        self.buffer = b""
+    client = None  # type: Client
+    transport = None  # type: asyncio.WriteTransport
+
+    def __init__(self, client: 'Optional[Client]' = None) -> None:
+        if client is not None:
```

Comment:
We'd explicitly initialize `self.client` with `Optional[Client]` value, but the variable is `Client`. We can

a) make `Protocol.client` `Optional` and do checks in all methods... I don't like it
b) let `Protocol.__init__` accept `Client` instead of `Optional[Client]` - this would require rewriting some text fixtures, right?
c) leave it as is :)